lib/gauche/regexp/sre.scm

```diff
@@ -176,6 +177,11 @@
         [(w/ascii) (seq-loop rest nocapture casefold #t)]
         [(w/unicode) (seq-loop rest nocapture casefold #f)]
         [(w/nocapture) (seq-loop rest #t)]
+        [(word) (%sre->ast `(: bow ,@rest eow)
+                           nocapture casefold ascii)]
+        [(word+) (%sre->ast `(word (+ (and (or "_" alphanumeric)
```

Comment:
We could export "WORD" charset and use it here instead of `(or "_" alphanumeric)`. But it's non-standard charset anyway, and "WORD" is limited to ascii, this is probably better.